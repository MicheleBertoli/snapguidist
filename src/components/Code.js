import React from 'react'
import PropTypes from 'prop-types'
import Codemirror from 'react-codemirror'
import 'codemirror/mode/diff/diff'
import 'codemirror/mode/jsx/jsx'

const Code = (props, context) => {
  const options = {
    mode: props.diff ? 'diff' : 'jsx',
    theme: context.config.highlightTheme,
  }

  return (
    <div>
      <div className="snapguidist__label">{props.label}</div>
      <div className="snapguidist__code">
        <Codemirror value={props.value} options={options} />
      </div>
    </div>
  )
}

Code.propTypes = {
  diff: PropTypes.bool,
  label: PropTypes.string,
  value: PropTypes.string,
}

Code.contextTypes = {
  config: PropTypes.object.isRequired,
}

export default Code
