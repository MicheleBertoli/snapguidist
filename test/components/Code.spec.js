import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Code from '../../src/components/Code'

const props = {
  diff: false,
  label: 'label',
  value: 'value',
}
const context = { config: { highlightTheme: 'theme' } }

test('works with jsx', () => {
  const wrapper = shallow(<Code {...props} />, { context })

  expect(toJSON(wrapper)).toMatchSnapshot()
})

test('works with diff', () => {
  const wrapper = shallow(<Code {...props} diff />, { context })

  expect(toJSON(wrapper)).toMatchSnapshot()
})
