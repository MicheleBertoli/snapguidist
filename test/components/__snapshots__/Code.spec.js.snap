// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with diff 1`] = `
<div>
  <div
    className="snapguidist__label"
  >
    label
  </div>
  <div
    className="snapguidist__code"
  >
    <CodeMirror
      options={
        Object {
          "mode": "diff",
          "theme": "theme",
        }
      }
      value="value"
    />
  </div>
</div>
`;

exports[`works with jsx 1`] = `
<div>
  <div
    className="snapguidist__label"
  >
    label
  </div>
  <div
    className="snapguidist__code"
  >
    <CodeMirror
      options={
        Object {
          "mode": "jsx",
          "theme": "theme",
        }
      }
      value="value"
    />
  </div>
</div>
`;
